File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        stack<char> st;
        string ans = "";
        int cnt = 0;
        for(char ch:s){
            if(ch == '(')
            {
                if(!st.empty())
                    ans += ch;
                st.push(ch);
            }
            else{
                st.pop();
                if(!st.empty())
                    ans += ch;
            }
        }
        return ans;
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    static {
        for (int i = 0; i <= 500; i++) {
            removeOuterParentheses("");
        }
    }

    public static String removeOuterParentheses(String s) {
        int cnt = 0;
        StringBuilder sb = new StringBuilder();
        for (char ch : s.toCharArray()) {
            if (ch == '(') {
                if (cnt > 0)
                    sb.append('(');
                cnt++;
            } else {
                cnt--;
                if (cnt > 0)
                    sb.append(')');
            }
        }
        return sb.toString();
    }
}